### PR TITLE
Odfv logic

### PR DESCRIPTION
**Pull request by felixwang9817**
**What this PR does / why we need it**: This PR refactors some of the ODFV logic in `get_online_features`.

**Which issue(s) this PR fixes**:

Fixes #

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

File: sdk/python/feast/feature_store.py [hunk 1]
```diff
@@ -1238,28 +1235,49 @@ def _augment_response_with_on_demand_transforms(
...
+                changing them from the format "feature" to "feature_view__feature" (e.g., "daily_transactions" changes to
+                "customer_fv__daily_transactions").
+            result_rows: List of result rows to be augmented with on demand feature values.
+        """
+        requested_odfv_map = {
+            odfv.name: odfv for odfv in requested_on_demand_feature_views
+        }
+        requested_odfv_feature_names = requested_odfv_map.keys()
+
+        if len(requested_odfv_map) == 0:
```
Comment (adchia) [inline sdk/python/feast/feature_store.py:1263]: nit: pull this check first and check requested_on_demand_feature_views instead?

**Review by adchia — CHANGES_REQUESTED** [review]
/lgtm

**Comment by adchia** [comment]
Also, I think this fixes a bug too right? Mind tagging that issue?

**Review by achals — APPROVED** [review]
/lgtm

**Pull request merged**